protocols/imap/src/main/java/org/apache/james/imap/processor/AbstractMailboxProcessor.java

```diff
@@ -83,9 +88,12 @@ protected final void doProcess(M acceptableMessage, Responder responder, ImapSes
     }
 
     protected final void process(M message, Responder responder, ImapSession session) {
-        final ImapCommand command = message.getCommand();
-        final String tag = message.getTag();
+        ImapCommand command = message.getCommand();
+        String tag = message.getTag();
+
+        TimeMetric timeMetric = metricFactory.timer(command.getName());
         doProcess(message, command, tag, responder, session);
+        timeMetric.elapseTimeInMs();
```

Comment:
Ok I still think that when I read a method name such as `elapseTimeInMs` which should be `elapsedTimeInMs` by the way. It look like a pure function to me that just return the elapsedTimeInMs.
It would be great to have a name that let know there is a side effect no matter what it is maybe something such as:
timeMetric.stopTimewatchAndGetElapsedTimeInMs would be better.
Moreover, you say that elapseTimeInMs stop the timewatch when called, I did not understood that before you told it, I thought it would just let the timewatch running and return the current elapsed time